griptape/tools/image_query_client/tool.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from typing import Any, cast
+
+from attr import define, field, Factory
+from schema import Schema, Literal
+
+from griptape.artifacts import TextArtifact, ImageArtifact, ErrorArtifact
+from griptape.loaders import ImageLoader
+from griptape.tools import BaseTool
+from griptape.utils import load_artifact_from_memory
+from griptape.utils.decorators import activity
+from griptape.engines import ImageQueryEngine
+
+
+@define
+class ImageQueryClient(BaseTool):
+    image_query_engine: ImageQueryEngine = field(kw_only=True)
+    image_loader: ImageLoader = field(default=Factory(lambda: ImageLoader()), kw_only=True)
+
+    @activity(
+        config={
+            "description": "This tool can be used to query the contents of images on disk.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "query",
+                        description="A detailed question to be answered using the contents of the provided images.",
+                    ): str,
+                    Literal("image_paths", description="The paths to an image files on disk."): list[str],
+                }
+            ),
+        }
+    )
+    def query_image_from_disk(self, params: dict) -> TextArtifact | ErrorArtifact:
```

Comment:
not sure if this is a pattern we've used yet: but should this be using the file_manager tool? it has a whole advanced interface for picking the proper loader and error checking for it.

Reply:
This pattern is used in text+image to image generation tools like VariationImageGenerationClient, InpaintingImageGenerationClient, and OutpaintingImageGenerationClient to provide some flexibility on the source of the input images. I don't think FileManager would make sense here because we only want to load images, so we don't need that loader selection logic.

Reply:
That's fair. It's just using the ImageLoader. I was more thinking on the validation side. Are there any validations we need to do here to make sure the user is handing us an image file? does it just blow up if it's not an image file?

Reply:
discussed in person. the underlying parser will properly surface errors if there are problems parsing.